lib/puppet/parser/functions/pick.rb

```diff
@@ -15,9 +15,22 @@ module Puppet::Parser::Functions
     called 'jenkins_version' (note that parameters set in the Puppet Dashboard/
     Enterprise Console are brought into Puppet as top-scope variables), and,
     failing that, will use a default value of 1.449.
+
+    If you have `strict_variables` turned on, then wrap your variable in single
+    quotes to prevent interpolation and this function will check to see if that
+    variable exists.
+
+      $real_jenkins_version = pick('$::jenkins_version', '1.449')
+
 DOC
              ) do |args|
-    args = args.compact
+
+    # look up the values of any strings that look like '$variables'
+    args.map! do |item|
```

Comment:
Should absolutely not mutate the given args!! Must work on a copy. earlier the `args.compact` provided the copy now this logic will destroy the caller's array of arguments.